npm/webpack-batteries-included-preprocessor/test/e2e/features.spec.js

```diff
@@ -0,0 +1,119 @@
+const EventEmitter = require('events').EventEmitter
+const { expect } = require('chai')
+const fs = require('fs-extra')
+const path = require('path')
+
+const preprocessor = require('../../index')
+
+const fixturesDir = path.join(__dirname, '..', 'fixtures')
+const outputDir = path.join(__dirname, '..', '_test-output')
+
+const run = (fileName, options) => {
+  const file = Object.assign(new EventEmitter(), {
+    filePath: path.join(outputDir, fileName),
+    outputPath: path.join(outputDir, fileName.replace('.', '_output.')),
+  })
+
+  return preprocessor(options)(file)
+}
+
+const runAndEval = async (fileName, options) => {
```

Comment:
I haven't seen this kind of test suite for a webpack plugin before, this is cool - good reference for when we need to write some plugins for our webpack plugin for the component testing architecture